crates/astria-cli/src/commands/bridge.rs

```diff
@@ -0,0 +1,484 @@
+            bridge_address: self.bridge_address,
+            sequencer_address_prefix: self.sequencer_address_prefix.clone(),
+        }
+        .try_convert()
+        .wrap_err("failed converting log to ics20 withdrawal action")
+    }
+
+    fn log_to_sequencer_withdrawal_action(&self, log: Log) -> eyre::Result<Action> {
+        LogToSequencerWithdrawalAction {
+            log,
+            fee_asset: self.fee_asset.clone(),
+            asset_withdrawal_divisor: self.asset_withdrawal_divisor,
+        }
+        .try_into_action()
+        .wrap_err("failed converting log to sequencer withdrawal action")
+    }
+}
+
+fn action_inputs_from_log<T: EthEvent>(log: Log) -> eyre::Result<(T, u64, [u8; 32])> {
+    let block_number = log
+        .block_number
+        .ok_or_eyre("log did not contain block number")?
+        .as_u64();
+    let transaction_hash = log
+        .transaction_hash
+        .ok_or_eyre("log did not contain transaction hash")?
+        .into();
+
+    let event = T::decode_log(&log.into())
+        .wrap_err_with(|| format!("failed decoding contract log as `{}`", T::name()))?;
+    Ok((event, block_number, transaction_hash))
+}
+
+#[derive(Debug)]
+struct LogToIcs20WithdrawalAction {
+    log: Log,
+    fee_asset: asset::Denom,
+    rollup_asset_denom: asset::Denom,
+    asset_withdrawal_divisor: u128,
+    bridge_address: Address,
+    sequencer_address_prefix: String,
+}
+
+impl LogToIcs20WithdrawalAction {
+    fn try_convert(self) -> eyre::Result<Action> {
+        let Self {
+            log,
+            fee_asset,
+            rollup_asset_denom,
+            asset_withdrawal_divisor,
+            bridge_address,
+            sequencer_address_prefix,
+        } = self;
+
+        let (event, block_number, transaction_hash) =
+            action_inputs_from_log::<Ics20WithdrawalFilter>(log)
+                .wrap_err("failed getting required data from log")?;
+
+        let sender = event.sender.to_fixed_bytes();
+
+        let source_channel = rollup_asset_denom
+            .as_trace_prefixed()
+            .and_then(TracePrefixed::last_channel)
+            .ok_or_eyre("rollup asset denom must have a channel to be withdrawn via IBC")?
+            .parse()
+            .wrap_err("failed to parse channel from rollup asset denom")?;
+
+        let memo = Ics20WithdrawalFromRollupMemo {
+            memo: event.memo,
+            bridge_address,
+            block_number,
+            transaction_hash,
+        };
+
+        let action = Ics20Withdrawal {
+            denom: rollup_asset_denom,
+            destination_chain_address: event.destination_chain_address,
+            // note: this is actually a rollup address; we expect failed ics20 withdrawals to be
+            // returned to the rollup.
+            // this is only ok for now because addresses on the sequencer and the rollup are both 20
+            // bytes, but this won't work otherwise.
```

Comment:
so this is not applicable anymore - we need to also fix this in the withdrawer. this will return funds back to the sequencer address, but we want them to be refunded back to the rollup. we can leave this to a follow-up. basically this should be set to the sender of the withdrawal tx on the rollup, which you should be able to extract this from the transaction data.

Reply:
So this is actually rollup-specific data, not sequencer-specific?

Should we then make this a breaking change and coll this field `bytes rollup_return_address = x;` in the protobuf?

Reply:
Done in [8333ecf](https://github.com/astriaorg/astria/pull/1261/commits/8333ecfad8bec2d48d213fd27234242e3f1638dc) in after merging https://github.com/astriaorg/astria/pull/1266